synapse/storage/user_directory.py

```diff
@@ -569,7 +642,7 @@ def search_user_dir(self, user_id, search_term, limit):
         else:
             join_clause = """
                 LEFT JOIN (
-                    SELECT other_user_id AS user_id FROM users_who_share_public_rooms
+                    SELECT user_id FROM users_in_public_rooms
```

Comment:
I haven't tested it yet but I'm fairly sure this isn't going to optimise correctly. Can we revert to the previous method of using `LEFT JOIN` which did work correctly?

Reply:
done